sosed/run.py

```diff
@@ -71,8 +75,72 @@ def vectorize(processed_data: ProcessedData, force: bool) -> None:
     processed_data.store_repo_stats(all_repo_stats)
 
 
+def analyze_topics(
+        processed_data: ProcessedData, output_dir: str
+) -> None:
+    repo_vectors = probability_vectors(smooth_vectors(processed_data.load_repo_vectors()))
+    cnts = processed_data.load_repo_vectors().sum(axis=1, keepdims=True)
+    repo_names = processed_data.load_repo_names()
+    clusters_info = get_clusters_info()
+    out_file_path = os.path.join(output_dir, f"topics.json")
+    with open(os.path.abspath(out_file_path), "w+") as fout:
+        is_first_repo = True
+        repo_name = ""
+        for file_name, repo_vector, cnt in zip(repo_names, repo_vectors, cnts):
+            if is_first_repo:
+                repo_name = file_name[:-1]
+                fout.write('{\n')
+                fout.write('\t\"data\" : [\n')
+                fout.write(f'\t\t{{\n')
+                fout.write(f'\t\t\t\"path\" : "{repo_name}\",\n')
+                fout.write(f'\t\t\t\"files\" : [\n')
+                is_first_repo = False
+            else:
+                if not file_name.startswith(repo_name):
+                    repo_name = file_name[:-1]
+                    fout.write('\n\t\t\t]\n')
+                    fout.write('\t\t},\n')
+                    fout.write('\t\t{\n')
+                    fout.write(f'\t\t\t\"path\" : "{repo_name}\",\n')
+                    fout.write(f'\t\t\t\"files\" : [\n')
+                else:
+                    fout.write(",\n")
+            fout.write(f'\t\t\t\t{{\n')
+            fout.write(f'\t\t\t\t\t\"path\" : "{file_name[len(repo_name):]}\",\n')
+            fout.write(f'\t\t\t\t\t\"topics\" : [')
+            topics = np.argsort(repo_vector)[-5:][::-1]
+            is_first = True
+            if cnt != 0:
+                for dim in topics:
+                    if is_first:
+                        is_first = False
+                        fout.write(f'\n')
+                    else:
+                        fout.write(f',\n')
+                    topic_name = clusters_info[dim][0].replace('"', "'")
+                    fout.write(f'\t\t\t\t\t\t\"{topic_name}\"')
+            fout.write(f'\n\t\t\t\t\t],\n')
+            fout.write(f'\t\t\t\t\t\"probs\" : [')
+            is_first = True
+            if cnt != 0:
+                for dim in topics:
+                    if is_first:
+                        is_first = False
+                        fout.write(f'\n')
+                    else:
+                        fout.write(f',\n')
+                    fout.write(f'\t\t\t\t\t\t\"{repo_vector[dim]:.3f}\"')
+            fout.write(f'\n\t\t\t\t\t]\n')
+            fout.write('\t\t\t\t}')
+        fout.write('\n\t\t\t]\n')
+        fout.write('\t\t}\n')
+        fout.write('\t]\n')
+        fout.write('}\n')
+    print(f'JSON with tokens was written in {out_file_path}.')
```

Comment:
If you are writing a JSON here, consider using a built-in library `json` to write it (you can toggle on indents and other formatting if you want).